docs/types.rst

```diff
@@ -52,6 +52,12 @@ Operators:
 * Bit operators: ``&``, ``|``, ``^`` (bitwise exclusive or), ``~`` (bitwise negation)
 * Arithmetic operators: ``+``, ``-``, unary ``-``, unary ``+``, ``*``, ``/``, ``%`` (remainder), ``**`` (exponentiation), ``<<`` (left shift), ``>>`` (right shift)
 
+The bit size of an integer determines its range, for example, for ``uint256``, this is 0 up to 2 :sup:`256` -1. If the result of an operation doesn't fit inside this range, it is truncated. These truncations can have `serious consequences <https://en.bitcoin.it/wiki/Value_overflow_incident>`_, so use code like the below to avoid certain attacks and check a value is what you expect it to be.
+
+::
+
+    require((balanceOf[_to] + _value) >= balanceOf[_to]);
+
```

Comment:
Should we have a note stating "or use audited libraries providing this functionality"?

Also, should we leave a note that in the future this may change with having runtime overflow checking?

Lastly, should it link to the (non-existent) SMT section? :)

Reply:
@axic I can't speak too much for the non-existent section right now, I think I saw a PR about it, so will note that todo. If we mention using audit libraries, I'd love to say how to find those, any resources?

As for things that might happen in the future, I generally think it's best to mention that when it's actually a feature to be honest, as it may never happen, and then the docs look odd.